caom2utils/caom2utils/tests/test_fits2caom2.py

```diff
@@ -1185,8 +1189,35 @@ def test_visit_generic_parser():
                     'test_collection_id', 'test_observation_id']
         test_parser = GenericParser()
         test_plugin = __name__
-        test_args = get_gen_proc_arg_parser().parse_args()
         kwargs = {}
-        _visit(test_plugin, test_args, test_parser, **kwargs)
+        test_obs = SimpleObservation(collection='test_collection',
+                                     observation_id='test_obs_id',
+                                     algorithm=Algorithm(str('exposure')))
+        _visit(test_plugin, test_parser, test_obs, visit_local=None, **kwargs)
     except BaseException as e:
         assert False, 'should not get here {}'.format(e)
+
+
+@pytest.mark.skipif(single_test, reason='Single test mode')
```

Comment:
I think the vos Client class should be mocked if it's not.

Reply:
Done.